src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -588,8 +589,18 @@ public FormulaType GetNodeType(TexlNode node)
 
         // Called by language server to get custom tokens.
         // If binding is available, returns context sensitive tokens.  $$$
+        // Keeping this temporarily for custom publish tokens notification
+        // Feature to auto fix casing of function name in the editor also depends on this function and custom publish tokens notification
         internal IReadOnlyDictionary<string, TokenResultType> GetTokens(GetTokensFlags flags) => GetTokensUtils.GetTokens(this._binding, flags);
 
+        /// <summary>
+        /// Returns an enumeration of token text spans in a expression rule with their start and end indices and token type.
+        /// </summary>
+        /// <param name="comparer">Optional comparer to sort tokens.</param>
+        /// <param name="allowTokenHiding">Optional flag to indicate whether to compute whether token can be hidden or not.</param>
+        /// <returns>Ordered or undordered collection of tokens.</returns>

```

Comment:
>Ordered or undordered  [](http://example.com/codeflow?start=21&length=22)

nit: phrasing?  just "unordered"  as is, not necessarily sorted by span.

Reply:
Yes. The idea is to not sort the tokens at all if comparer is not provided. The reason why i did it this way is language server doesn't need to spend probably nlogn time to sort the tokens as they will be encoded in the order based on the spans. The algorithm to encode goes from the smallest possible start index (0) to the biggest possible length (expression.Length - 1). Since we are iterating in the sorted order already, why spend time to sort it unnecessarily. If someone wants to use this function for other purposes, they can pass in a comparer that sorts based on start or end indexes. Assuming the number of tokens are <= expression.Length, assuming that C# sort is no better than nlogn, if we sort by spans and then encode the computational cost would be Sorting time = O(expression.Length * Log(expression.Length) + encoding time = O(expression.Length). If we just encode and given that we traverse expression from left to right, we already processing tokens in sorted order so computational cost would be O(expression.Length)

Reply:
Resolving this. Let me know if you have any feedback around this